### DISCUSSION TITLE
log access login date

### DISCUSSION THREAD

**Discussion by teafmartins** [serenity-is/Serenity#6329] [Help] [answered]
Hi,

I need to register users access login date.
Where in serenity should I include this log routine?

Thx

**Comment by teafmartins** [comment, marked as answer]
in AuthenticationService.cs

...
            if (user.Source == "site" || user.Source == "sign" || directoryService == null)
            {
                if (validatePassword())
                {
                    **// Log it here**
                    throttler.Reset();
                    return true;
                }
..

**Reply by minhhungit** [reply]
yes, in AuthenticationService.cs but I think better you should log it in **Validate** function
If you need to log detail user info, add log in both ValidateExistingUser and ValidateFirstTimeUser

**Reply by teafmartins** [reply]
why ValidateExistingUser and ValidateFirstTimeUser ?
ValidateExistingUser  is not enough?

**Reply by minhhungit** [reply]
ValidateFirstTimeUser is for the case that user has not existsed, maybe when user is registried

**Reply by teafmartins** [reply]
that's it! :) thanks for your time
